src/kehaar_component/external_service.clj

```diff
@@ -0,0 +1,54 @@
+(ns kehaar-component.external-service
+  (:require
+    [clojure.core.async :as async]
+    [com.stuartsierra.component :as component]
+    [kehaar.wire-up :as wire-up]
+    [langohr.core :as rmq]))
+
+;; Pseudo spec/schema just so it's clear what we expect for config
+;; here--could later be used to validate config with some adjustments.
+
+(def config-spec
+  {:type :external
+   :queue-name "external-service-queue-name"
+
+   ;; Kehaar/RabbitMQ config
+   :exclusive false
+   :durable true
+   :auto-delete false})
+
+(defprotocol IExternalService
+  (call-external! [this msg]))
+
+(defrecord ExternalService [rabbitmq config msg-chan service call-external]
+  component/Lifecycle
+
+  (start [component]
+    (println ";; Starting ExternalService " (:queue-name config))
+
+    (let [{:keys [queue-name timeout]} config
+          msg-chan (async/chan (or timeout 1000))
```

Comment:
Why is `timeout` determining the buffer size? Should there be a different key for buffer size and the timeout used below in the call to `wire-up/external-service`?